xla/service/gpu/fusions/triton/triton_fusion_emitter_legacy_matmul.cc

```diff
@@ -1400,7 +1433,7 @@ class MatMulEmitterHelper {
     if (dims_.out_split_k_dim_idx.has_value()) {
       const TensorIterationSpec::DimIterationSpec* spec = analysis_.IterSpec(
           TritonFusionAnalysis::Scope::OUTPUT, hlo, *dims_.out_split_k_dim_idx);
-      if (spec != nullptr) {
+      if (spec != nullptr && spec->at(0).count > 1) {
```

Comment:
What changes in the `split-k` case?

Reply:
With split-K in the epilogue of the fusion there are input tensors which do not have the split-K-batch dimension and output tensors which do. Only the output ones should be iterated over along this dimension. EmitTensorPointer() is used for both. Before this change the split-K-batch dimension would never appear at the input tensor iteration specs, so it was enough to check only for its presence. Now it can appear, with a trivial spec (count=1), because fused broadcasts with trivial dimensions are now supported.

Reply:
The added test BroadcastsOfTriviallySizedNonContractingDimensionsAreSupported will fail without this change, demonstrating the problem.